javacutil/src/main/java/org/checkerframework/javacutil/PluginUtil.java

```diff
@@ -34,11 +34,31 @@
     public static final String JAVAC_PATH_OPT = "-javacJar";
 
     /**
-     * Option name for specifying an alternative jdk.jar location. The accompanying value MUST be
-     * the path to the jar file (NOT the path to its encompassing directory)
+     * Option name for specifying an alternative location for the annotated JDK.
+     *
+     * <p>In case of Java 8, an alternative jdk.jar location. The accompanying value MUST be the
+     * path to the jar file (NOT the path to its encompassing directory).
+     *
+     * <p>In case of Java 9+, a directory containing annotated &lt;module&gt;.jar. The supplied
```

Comment:
Instead of `.jar` should this be `.jmod`?

Reply:
https://docs.google.com/document/d/1mcOJUrYzHDMp2MfPztOmOaaSFjOWZquUN1fO_LCmXuo/edit?disco=AAAACPnypBE

Reply:
Can you be a bit more specific? That document states "We will have replacement .jmod files, e.g. an annotated java.base.jmod for most of our existing annotations".
So why is this directory containing `.jar` files instead of `.jmod` files?

Reply:
We can use both `.jmod` and `.jar` files for `--patch-module` argument. Since it is easier to create `.jar` file I went with this option.
I also discussed this with @mernst when I was first starting and he agrees with me.
Is there any advantage to use `.jmod`?

Reply:
More precisely, I didn't know a reason to object to .jar files.
@wmdietl are .jmod files better?

Reply:
The normal OpenJDK build creates .jmod files. If you follow the [normal build instructions](https://github.com/AdoptOpenJDK/openjdk-jdk/blob/master/doc/building.md), you get all the .jmod files in `build/linux-x86_64-server-release/images/jmods/` or some similar directory.
So instead of asking why .jmod files are better, why should we add additional build steps and create separate .jar files?

Reply:
@the1derer , can you explain your comment that it's easier to create the `.jar` files?
Sticking to the normal build process would be good, if possible.

Reply:
@mernst @wmdietl Apologies for the late reply. I was checking how to use JMOD files.

I agree sticking to the normal build process would be good, but after compiling (annotated)JDK11 I tried using JMOD files in `build/linux-x86_64-server-release/images/jmods/` via `--patch-module java.base=java.base.jmod` but it didn't seem to work(ie. it is still issuing,`warning: You do not seem to be using the distributed annotated JDK.`) on the other hand, using `build/linux-x86_64-server-release/jdk/modules`(from which I created my JAR files) seem to work fine.
I am not able to understand why this problem is occurring.

Reply:
In other words:  --patch-module works with jar files but not with jmod files.